packages/build-utils/test/unit.framework-detector.test.ts

```diff
@@ -40,115 +47,286 @@ class VirtualFilesystem extends DetectorFilesystem {
+                  : [name, subDirectoryName].join('/'),
+              type: 'dir',
+            };
+          }
+
+          return null;
+        })
+        // remove nulls
+        .filter((stat): stat is Stat => stat !== null)
+        // remove duplicates
+        .filter(
+          (stat, index, self) =>
+            index ===
+            self.findIndex(s => s.name === stat.name && s.path === stat.path)
+        )
+    );
+  }
+
+  /**
+   * An example of how to implement chdir for a virtual filesystem.
+   */
+  _chdir(name: string): DetectorFilesystem {
+    const basePath = path.join(this.cwd, name).replaceAll(path.sep, '/');
+    const files = Object.fromEntries(
+      [...this.files.keys()].map(key => [key, this.files.get(key) ?? ''])
+    );
+
+    return new VirtualFilesystem(files, basePath);
+  }
 }
 
-describe('#detectFramework', () => {
-  it('Do not detect anything', async () => {
-    const fs = new VirtualFilesystem({
-      'README.md': '# hi',
-      'api/cheese.js': 'export default (req, res) => res.end("cheese");',
-    });
+describe('DetectorFilesystem', () => {
+  it('should return the directoy contents relative to the cwd', async () => {
+    const files = {
+      'package.json': '{}',
+      'packages/app1/package.json': '{}',
+      'packages/app2/package.json': '{}',
+    };
 
-    expect(await detectFramework({ fs, frameworkList })).toBe(null);
+    const fs = new VirtualFilesystem(files);
+
+    expect(await fs.readdir('/')).toEqual([
+      { name: 'package.json', path: 'package.json', type: 'file' },
+      { name: 'packages', path: 'packages', type: 'dir' },
+    ]);
+
+    expect(await fs.readdir('packages')).toEqual([
+      { name: 'app1', path: 'packages/app1', type: 'dir' },
+      { name: 'app2', path: 'packages/app2', type: 'dir' },
+    ]);
+
+    expect(await fs.readdir('packages/app1')).toEqual([
+      {
+        name: 'package.json',
+        path: 'packages/app1/package.json',
+        type: 'file',
+      },
+    ]);
   });
 
-  it('Detect Next.js', async () => {
-    const fs = new VirtualFilesystem({
-      'package.json': JSON.stringify({
-        dependencies: {
-          next: '9.0.0',
-        },
-      }),
+  it('should be able to change directories', async () => {
+    const nextPackageJson = JSON.stringify({
+      dependencies: {
+        next: '9.0.0',
+      },
     });
+    const gatsbyPackageJson = JSON.stringify({
+      dependencies: {
+        gatsby: '1.0.0',
+      },
+    });
+
+    const files = {
+      'package.json': '{}',
+      'packages/app1/package.json': nextPackageJson,
+      'packages/app2/package.json': gatsbyPackageJson,
+    };
+
+    const fs = new VirtualFilesystem(files);
+    const packagesFs = fs.chdir('packages');
+
+    expect(await packagesFs.readdir('/')).toEqual([
+      { name: 'app1', path: 'app1', type: 'dir' },
+      { name: 'app2', path: 'app2', type: 'dir' },
+    ]);
+
+    expect(await packagesFs.hasPath('app1')).toBeTruthy();
```

Comment:
Nit pick: these should be `.toBe(true)` because it should return a boolean every time